book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+
+### READ
+
+| Custom Resources | When? | Why? |
+|---|---|---|
+| Application & Component | Before creating the Snapshots | To know which build.ImageSpecs to use in the Snapshot |
+| IntegrationTestScenario | Before creating the Component/Composite PipelineRun(s) | To get the information for the PipelineRun - which Tekton bundles and Environments will be used |
+| ReleaseLink | Before creating the Release | To signal the Release Service for next environment promotion (prod) |
+
+### UPDATE
+
+| Custom Resource | When? | Why? |
+|---|---|---|
+| SnapshotEnvironmentBinding | Post Test PipelineRun | To promote the Snapshot to the next environment (Stage/Stable) |
+| Snapshot | Post Test PipelineRun | To mark the snapshot validated so that it can be promoted to the next environment |
+| Component | Post Test PipelineRun | To update the Component with the image spec, in order to update the global candidates list |
+
+### WATCH
+
+| Custom Resource | When? | Why? |
+|---|---|---|
+| PipelineRun | Post Build PipelineRun | Read the component and the output image to create the Snapshot and start the Preliminary PipelineRun |
+| PipelineRun | Post Preliminary PipelineRun | Check the result of the Preliminary PipelineRun and  follow up on the global candidate list state to either trigger the Final PipelineRun or check ReleaseLink and create a Release |
+| PipelineRun | Post Final PipelineRun | Check the result of the Final PipelineRun, check ReleaseLink and create a Release |
+
+### Pipeline
+
+The test pipeline will consist of a single definition that could be executed as two logically distinct pipeline runs by the Integration Service. The logical pipeline runs are:
+
+- Component (Prelim) pipeline run
+- Composite (Final) pipeline run
+
+The Component pipeline run is always triggered for each Component of an Application and is triggered by a completed build of a component.  
+
+The Composite run is only triggered when it has been determined by the Integration Service that two or more Components of an Application have been successfully built concurrently by their respective Component pipeline and need to be tested together. 
+
+The two stage model is roughly modeled on the 'fast loop / slow loop' model that OpenShift uses.  ART is the slow loop.  The global candidates list at some point in time is the set of artifacts that the Composite pipeline will test and mark as a 'shippable' snapshot.
+
+[Draft: StoneSoup-4 Pipeline & Task Design](https://docs.google.com/document/d/1ucQI3WJzkLpueYLuzHEXpKNvYQafDl4gNBZryvi2d5o/edit#)
+
+### PipelineRun
+
+The Integration Service will generate a PipelineRun that can reference the user provided tekton pipeline. 
+
+#### Pipeline Reference
+The pipeline reference should point to a well known location that an Application team can provide their tekton bundle. The pipeline packaging will be retrieved the IntegrationScenario CR
+
+#### Annotations/Labels
```

Comment:
I assume that this section is just stating what annotations/labels the service is setting. Instead of stating what "we" can or must do, just simply state what is being done.

Reply:
Yes, this is correct - all of this information is set by the Integration service.
This section is mostly directly useful to other Stone Soup workflows - some of this information will be bubbled up to the users by way of UI, and some power users might directly fetch it in the future.

Reply:
I addressed the specific usages of `we` in separate comments where appropriate.

Reply:
I would also suggest changing the following line

```
Following the [guidelines](https://docs.google.com/document/d/1gyXM3pkKFFfxHZnopBi_53vREFWhwA0pFUuIhopDuEo/edit#) in we can define some basic ones to start
```
to something like
```
Following the [annotation guidelines](https://docs.google.com/document/d/1gyXM3pkKFFfxHZnopBi_53vREFWhwA0pFUuIhopDuEo/edit#), the Integration Service sets the following annotations on PipelineRuns
```

Too bad that these google documents are not ADRs to be able to link to directly here. :-p